### PR TITLE
(bugfix) Fix asyncclient timeout param

### PR DESCRIPTION
**Pull request by thomasht86**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Closing issue #950. 
✅ Added unit tests to test that params are correctly passed through.
Also added `kwargs` that will be passed to async-client.

### CODE REVIEW

File: vespa/application.py [hunk 1]
```diff
@@ -1517,9 +1543,17 @@ def _open_httpx_client(self):
         limits = httpx.Limits(
             max_keepalive_connections=self.connections,
             max_connections=self.connections,
-            keepalive_expiry=10,  # This should NOT exceed the keepalive_timeout on the Server, otherwise we will get ConnectionTerminated errors.
+            keepalive_expiry=self.keepalive_expiry,  # This should NOT exceed the keepalive_timeout on the Server, otherwise we will get ConnectionTerminated errors.
         )
-        timeout = httpx.Timeout(pool=5, connect=5, read=5, write=5)
+        if self.total_timeout:
+            timeout = httpx.Timeout(self.total_timeout)
```
Comment (glebashnik) [inline vespa/application.py:1549]: If I understand the following code correctly this is not really total_timeout but more like default value for different timeouts:
https://github.com/encode/httpx/blob/1bf1fc0ea8417c8e539914aeafdb48f15591ba75/httpx/_config.py#L246
Perhaps we can let users pass Timeout object so we don't have to deal with the logic of how to set timeouts? This does break abstraction layer so kind of a tradeoff.

File: tests/unit/test_application.py [hunk 2]
```diff
@@ -693,5 +696,203 @@ def test_retry_on_429_status(self):
...
+    async def test_read_timeout(self):
+        app = Vespa(url="http://localhost", port=8080)
+        read_timeout = 1  # seconds
+        vespa_async = VespaAsync(app=app, total_timeout=None, read_timeout=read_timeout)
+
+        with patch(
+            "vespa.application.httpx.AsyncClient", autospec=True
+        ) as MockAsyncClient:
+            mock_client_instance = MockAsyncClient.return_value
+            mock_client_instance.post = AsyncMock(
```
Comment (glebashnik) [inline tests/unit/test_application.py:739]: Do we have a better way to simulate timeout than mock throwing an exception directly?

**Review by glebashnik — CHANGES_REQUESTED** [review]

File: vespa/application.py (hunk 1 above)
Comment (thomasht86) [reply vespa/application.py:1549]: Yes, good point. Tried to clarify in docstring for `total_timeout`-parameter. 
Did it this way to ensure backwards compatibility (by not changing parameter name). 
Good suggestion to let user pass Timeout. 
My suggestion is then to deprecate `total_timeout` parameter (issue `DeprecationWarning`,  and introduce `timeout: httpx.Timeout = httpx.Timeout(5)`. 
Does that sound reasonable?

Comment (glebashnik) [reply vespa/application.py:1549]: Sounds good.

File: tests/unit/test_application.py (hunk 2 above)
Comment (thomasht86) [reply tests/unit/test_application.py:739]: Hmm, agree that would be nice. I'll try to see if I can find a more relevant way.

**Review by glebashnik — APPROVED** [review]
Looks good

**Pull request merged**